File: archive/g/go/prime.go
package main

import (
	"fmt"
	"os"
	"strconv"
)

func main () {
	var IsPrime bool = true

	if len(os.Args) != 2 {
		fmt.Printf("Usage: %s [integer]\n", os.Args[0])
		os.Exit(1)
	}

	n, err := strconv.Atoi(os.Args[1])
	if err != nil {
		fmt.Printf("%s\n", err)
		fmt.Printf("Usage: %s [integer]\n", os.Args[0])
		os.Exit(1)
	}

	for i := 2; i <= n / 2; i++ {
		if n % i == 0 {
			IsPrime = false
			break
		}
	}

	if !IsPrime {
		fmt.Printf("Composite\n");
	} else {
		fmt.Printf("Prime\n");
	}
}

File: archive/r/r/prime.r
# Program to check if the input number is prime or not
args<-commandArgs(TRUE)
if(length(args) > 0){
  a1 = args[1]
  # Check Numeral only, ..
  numbers_only <- function(a) !grepl("\\D", a1)
  if(numbers_only(a1) == TRUE){
    if (a1 >= 0){
      a = as.integer(a1)
    if(a == 2){
      cat("Prime")
      # If 0, or 1 or Even Number
    }else if (a == 0 || a == 1 || a %% 2 == 0 ) {
      cat("Composite")
    }else{
      flag = 1
      r = a %/% 2
      for(i in 2:r) {
        if (a %% i == 0) {
           flag = 0
          break
          }
       }
       if(flag == 1) {
         cat("Prime")
        } else {
          cat("Composite")
        }
    }
  }else{# Empty Input
  cat("Usage: please input a non-negative integer")
}
}else{  # Negative Input
  cat("Usage: please input a non-negative integer")
}
}else{    # Empty Input
  cat("Usage: please input a non-negative integer")
}
